instrumentation/spring/spring-jms-2.0/javaagent/src/test/groovy/SpringListenerTest.groovy

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import io.opentelemetry.instrumentation.test.asserts.TraceAssert
+import io.opentelemetry.sdk.trace.data.SpanData
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import listener.AnnotatedListenerConfig
+import listener.ManualListenerConfig
+import org.springframework.context.annotation.AnnotationConfigApplicationContext
+import org.springframework.jms.core.JmsTemplate
+
+import javax.jms.ConnectionFactory
+
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER
+
+class SpringListenerTest extends AgentInstrumentationSpecification {
+  def "receiving message in spring listener generates spans"() {
+    setup:
+    def context = new AnnotationConfigApplicationContext(config)
+    def factory = context.getBean(ConnectionFactory)
+    def template = new JmsTemplate(factory)
+
+    template.convertAndSend("SpringListenerJms2", "a message")
+
+    expect:
+    assertTraces(2) {
+      traces.sort(orderByRootSpanKind(CONSUMER, PRODUCER))
+
+      trace(0, 1) {
+        consumerSpan(it, 0, "queue", "SpringListenerJms2", "", null, "receive")
+      }
+      trace(1, 2) {
+        producerSpan(it, 0, "queue", "SpringListenerJms2")
+        consumerSpan(it, 1, "queue", "SpringListenerJms2", "", span(0), "process")
+      }
+    }
+
+    cleanup:
+    context.close()
+
+    where:
+    config << [AnnotatedListenerConfig, ManualListenerConfig]
```

Comment:
the existing test (which passed with just JMS listener instrumentation) was `AnnotatedListenerConfig`, and the new test that exercises the new instrumentation is `ManualListenerConfig`